### PR TITLE
feat: Refactor Status to make it handle user presence requests

### PR DESCRIPTION
**Pull request by diegolmello**
## Proposed changes

- Status/index now calls getUserPresence instead of RoomItem
- Added RoomTypeIcon stories
- Added Status stories

## Issue(s)	

Depends on https://github.com/RocketChat/Rocket.Chat.ReactNative/pull/5162

## How to test or reproduce

- Make sure user presence (aka status) is working throughout the app
- This PR also fixes a bug:
  * go to a channel
  * tap on a user avatar you haven't seen yet (so you didn't load its presence)
  * app navigates to user info and fetches their status correctly (previously it would show as offline)

## Screenshots

image
image

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [ ] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

[NATIVE-371]

[NATIVE-371]: https://rocketchat.atlassian.net/browse/NATIVE-371?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/containers/RoomTypeIcon/index.tsx [hunk 1]
```diff
@@ -24,44 +25,38 @@ interface IRoomTypeIcon {
...
 
-	const color = themes[theme].titleText;
-	const iconStyle = [styles.icon, { color }, style];
+		const iconStyle = [styles.icon, style];
 
-	if (type === 'd' && !isGroupChat) {
-		if (!status) {
-			status = 'offline';
+		if (type === 'd' && !isGroupChat && userId) {
+			return <Status id={userId} style={iconStyle} size={size} status={status} />;
```
Comment (reinaldonetof) [inline app/containers/RoomTypeIcon/index.tsx:39]: Do you know if we still need to pass this status? If this prop `status` is been passed only for the omnichannel's room, isn't it?

File: app/containers/RoomItem/index.tsx [hunk 2]
```diff
@@ -124,7 +114,8 @@ const RoomItemContainer = React.memo(
 				autoJoin={autoJoin}
 				showAvatar={showAvatar}
 				displayMode={displayMode}
-				sourceType={item.source}
+				status={item.t === 'l' ? item?.visitor?.status : null}
```
Comment (GleidsonDaniel) [inline app/containers/RoomItem/index.tsx:117]:
```suggestion
				status={item.t === SubscriptionType.OMNICHANNEL ? item?.visitor?.status : null}
```
Even though the type is wrong, it's better to use the type than a string.

File: app/containers/RoomTypeIcon/OmnichannelRoomIcon.tsx [hunk 3]
```diff
@@ -49,7 +49,7 @@ export const OmnichannelRoomIcon = ({ size, style, sourceType, status }: IOmnich
 				<SvgUri
 					height={size}
 					width={size}
-					color={STATUS_COLORS[status || 'offline']}
+					color={STATUS_COLORS[status || 'offline'] ?? STATUS_COLORS.offline}
```
Comment (GleidsonDaniel) [inline app/containers/RoomTypeIcon/OmnichannelRoomIcon.tsx:52]: Is it really necessary to do this?

File: app/containers/Status/index.tsx [hunk 4]
```diff
@@ -1,21 +1,32 @@
...
 		return 'loading';
-	}) as TUserStatus;
-	return <Status size={size} style={style} status={status} {...props} />;
+	});
+
+	useEffect(() => {
+		if (connected && statusState === 'loading') {
+			getUserPresence(id);
+		}
+	}, [connected]);
```
Comment (GleidsonDaniel) [inline app/containers/Status/index.tsx:27]:
```suggestion
	}, [connected, statusState]);
```

**Review by GleidsonDaniel — CHANGES_REQUESTED** [review]

File: app/containers/RoomTypeIcon/index.tsx (hunk 1 above)
Comment (diegolmello) [reply app/containers/RoomTypeIcon/index.tsx:39]: It's going to be required on MessageComposer

File: app/containers/RoomTypeIcon/OmnichannelRoomIcon.tsx (hunk 3 above)
Comment (diegolmello) [reply app/containers/RoomTypeIcon/OmnichannelRoomIcon.tsx:52]: I did it for the stories, but I think I can remove it.

**Review by GleidsonDaniel — APPROVED** [review]

**Review by reinaldonetof — APPROVED** [review]

**Pull request merged**